### PR TITLE
fixing following a private library

### PR DESCRIPTION
**Pull request by ahsu1230**
@eishay @andrewconner @2is10 

Broken case today:
Friend B invites me (kifi user, not email) to his private library. I open up the library by email and try to follow. It'll be broken =(

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala [hunk 1]
```diff
@@ -1030,25 +1030,23 @@ class LibraryCommander @Inject() (
...
-      } else { // public or private library (called from kifi.com)
-        libraryInviteRepo.getWithLibraryIdAndUserId(libraryId, userId)
-      }
-      (lib, listInvites)
+      val tokenInvites = authToken.map { token =>
+        libraryInviteRepo.getByLibraryIdAndAuthToken(libraryId, token)
+      } getOrElse Seq.empty
+      val libInvites = libraryInviteRepo.getWithLibraryIdAndUserId(libraryId, userId)
+      val allInvites = tokenInvites ++ libInvites
+      (lib, allInvites)
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala:1041]: @eishay I think the idea behind your original PR was about right =) thanks for looking into this

**Comment by 2is10** [comment]
I’m not convinced it’s okay to use `authToken` without the passphrase here. An uninvited “attacker” with an `authToken` can easily construct this follow request and thereby gain access to the private library.

**Comment by ahsu1230** [comment]
so we don't always give a user the `hashedPassPhrase`. If friend B invites me (kifi user, not email) to his private library, the email I receive does not have the pass phrase. If I open the library by email, I can see the library without having to input a pass phrase. Is this right?

**Comment by 2is10** [comment]
Yes. In that case, we should ignore the `authToken` allow the follow because the logged-in user was invited to the library. This is the case that is broken today.

**Comment by 2is10** [comment]
I don’t think we should ever be giving out an `authToken` without a passphrase, btw.

**Comment by ahsu1230** [comment]
take back what I said about `hashedpassphrase`. It's still in, so the new code should also look for library invites with logged in userId.

If the library is private, but there are no library invites (with userId or with correct authToken/passphrase), then we return `Forbidden`

**Comment by 2is10** [comment]
This looks good to me. Thanks.

If convenient, it’d be good to add a test case that passes now but would fail with the old code, to help avoid a regression.

**Comment by ahsu1230** [comment]
Ok, I'll see what cases I can fit in. Thanks for looking it over!

**Pull request merged**